test/expr/closure/closures_swift6.swift

```diff
@@ -670,7 +756,7 @@ final class AutoclosureTests {
     doVoidStuff { [weak self] in
       doVoidStuff { [self] in
         guard let self else { return }
-        method() // expected-error {{call to method 'method' in closure requires explicit use of 'self' to make capture semantics explicit}}
```

Comment:
The explicit self version of this has been supported since [at least Swift 4.2](https://github.com/swiftlang/swift/pull/78738/files#r1921387676) so it seems perfectly fine for the implicit self version to be supported